### PR TITLE
Updated tasks

### PR DESCRIPTION
**Pull request by patricifrian**

### CODE REVIEW

File: Assignment4/patricifrian-190112/task07-updated.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+# Using RDFLib
+print("Print results using RDFLib")
+for s, p, o in g.triples((None, rdfs.subClassOf, ns.LivingThing)):
+    print(s)
+
+# Using SPARQL
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE {
+    ?Subject rdfs:subClassOf ns:LivingThing
```
Comment (dgarijo) [inline Assignment4/patricifrian-190112/task07-updated.py]: this does not retrieve all subclasses of subclasses of living thing

File: Assignment4/patricifrian-190112/task07-updated.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+# TO DO
+
+# Using RDFLib
+print("Print results using RDFLib")
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
+  print(s)
+for sub, p, o in g.triples((None,RDFS.subClassOf,ns.Person)):
+  for s, p, o in g.triples((None,RDF.type,sub)):
+    print(s)
```
Comment (dgarijo) [inline Assignment4/patricifrian-190112/task07-updated.py:67]: same as above

File: Assignment4/patricifrian-190112/task07-updated.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+rdfs = Namespace("http://www.w3.org/2000/01/rdf-schema#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+# Using RDFLib
+print("Print results using RDFLib")
+for s, p, o in g.triples((None, rdfs.subClassOf, ns.LivingThing)):
```
Comment (dgarijo) [inline Assignment4/patricifrian-190112/task07-updated.py:34]: same as below

File: Assignment4/patricifrian-190112/task07-updated.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+      FILTER (?known1 != ?known2)
+    }
+  }
+  GROUP BY ?entity
+  ''',
+  initNs={"foaf": foaf, "ns": ns}
+)
+# Visualize the results
+print("Entities who know at least two others:")
+for r in g.query(q1):
```
Comment (dgarijo) [inline Assignment4/patricifrian-190112/task07-updated.py:183]: missing last exercise in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address remaining comments

File: Assignment4/patricifrian-190112/task07-updated.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+for s, p, o in g.triples((None, rdfs.subClassOf, ns.LivingThing)):
+    print(s)
+    for q, p, o in g.triples((None, rdfs.subClassOf, s)):
+      print("Subclass:"+q)
+
+# Using SPARQL
+q1 = prepareQuery('''
+  SELECT ?Subject ?Subclass WHERE {
+    ?Subject rdfs:subClassOf ns:LivingThing .
+    OPTIONAL {?Subclass rdfs:subClassOf ?Subject .}
```
Comment (dgarijo) [inline Assignment4/patricifrian-190112/task07-updated.py:43]: what about the subclasses of subclasses of subclasses? You need to design a generic query.

**Review by dgarijo — CHANGES_REQUESTED** [review]
please address also old issues

**Comment by anaigmo** [comment]
Entregado con errores sin corregir, no se aceptan más PR

**Pull request merged**